integrations/adobe-analytics/lib/index.js

```diff
@@ -139,9 +147,90 @@ AdobeAnalytics.prototype.initialize = function() {
   // WITHOUT sending several 'Video Content Playing' events. (see line 1242)
   window._segHBPlayheads = {};
 
-  // Load the larger Heartbeat script only if the customer has it enabled in settings.
-  // This file is considerably bigger, so this check is necessary.
-  if (options.heartbeatTrackingServerUrl) {
+  // Start checking which SDK initialization to do
+  // Load the more compact Chromecast SDK only if the customer has it enabled in settings
+  if (options.chromecastMode) {
+    if(this.options && this.options.contextValues){
+    window.settingsContextValues = this.options.contextValues;
+    }
+    window.ADBmobile = {};
+    window.ADBmobile.analytics = {};
+    window.ADBMobileConfig = {
+      "marketingCloud": {
+        "org": `${options.marketingCloudOrgId}`
+      },
+      "target": {
+        "clientCode": "",
+        "timeout": 5
+      },
+      "audienceManager": {
+        "server": `${options.trackingServerUrl}`
+      },
+      // TO DO: Update placeholder values with settings or properties
+      "analytics": {
+        "rsids": `${options.reportSuiteId}`,
+        "server": `${options.trackingServerUrl}`,
+        "ssl": this.options.ssl,
+        "offlineEnabled": false,
+        "charset": "UTF-8",
+        "lifecycleTimeout": 300,
+        "privacyDefault": "optedin",
+        "batchLimit": 0,
+        "timezone": "MDT",
+        "timezoneOffset": -360,
+        "referrerTimeout": 0,
+        "poi": []
+      },
+      // TO DO: Update placeholder values with settings or properties
+      "mediaHeartbeat": {
+        "server": `${options.heartbeatTrackingServerUrl}`,
+        "publisher": `${options.marketingCloudOrgId}`,
+        "channel": "test-channel-chromecast",
```

Comment:
@Isuru-F any thoughts on if / how we could dynamically populate this? I think Fox only passes `channel` and `playerName` in the events, and I don't think those are available at this point. I also don't want to add a new setting in order to pipe something in. I can ask Fox if it matters what these values are for them, since the event metadata should indicate things like channel, player name, etc.? Do you have any indication on the point of these fields in the `mediaHeartbeat` config object?

Reply:
@kdaswani  - from what I can tell this JSON object looks like it must exist in a script tag on init, I don't think its something that can be passed in the track calls. 

That said is channel/playername something that should be "hardcoded" to chromecast? Looking at the definition in the docs it says
["String that represents the name of the content distribution channel." / "String that represents the name of the player."](https://experienceleague.adobe.com/docs/media-analytics/using/sdk-implement/setup/set-up-chromecast.html?lang=en)

If it does need to be dynamically set we may need to revisit a few things and do some more testing on a few parts to see how it behaves without it on init. 

For reference this is a chromecast AA example app I look at for some of these questions https://github.com/Adobe-Marketing-Cloud/media-sdks/blob/master/sdks/chromecast/samples/BasicPlayerSample/index.html

Reply:
Thanks @Isuru-F that was my sense too. Since it's in the initialization script, we won't have track calls yet to set values dynamically based on properties. I'm not super clear on what `channel` and `playerName` need to be (and how important they are) so I'm going to ask Fox for a copy of their existing Chromecast `ADBMobileConfig` since they integrate natively today. Hoping I can get that sorted with them next week and then we can consider this ready for testing. I'll keep you updated. Thanks for all your help!

Reply:
Chatted with Fox. They don't report on these fields and are not concerned about falling back on "chromecast" or "unknown" for these since we do that in the web integration: https://github.com/segmentio/analytics.js-integrations/blob/72e95388ce84f56d1776a8e2e546eba984def318/integrations/adobe-analytics/lib/index.js#L1232. Hardcoding `channel` sdkVersion` and `playerName` as a result.